ADR/0040-worspace-design-proposal.md

```diff
@@ -0,0 +1,172 @@
+# 40. Workspace Abstraction
```

Comment:
Can we change this ADR to 42? I am trying to deconflict the in-review ADRs. 

I think that we should still merge this ADR even if the status is not accepted.